blocks/bricks/bn.py

```diff
@@ -237,23 +248,27 @@ def _allocate(self):
         # aren't in self.parameters.
         add_annotation(self.population_mean, self)
 
-        if not self.mean_only:
+        if self.learn_scale and not self.mean_only:
             # "gamma", from the Ioffe & Szegedy manuscript.
             self.scale = shared_floatx_nans(var_dim, name='batch_norm_scale',
                                             broadcastable=broadcastable)
 
             add_role(self.scale, BATCH_NORM_SCALE_PARAMETER)
             self.parameters.append(self.scale)
+        else:
+            self.scale = tensor.constant(1., dtype=theano.config.floatX)
 
+        if not self.mean_only:
             self.population_stdev = shared_floatx(numpy.ones(var_dim),
                                                   name='population_stdev',
                                                   broadcastable=broadcastable)
             add_role(self.population_stdev, BATCH_NORM_POPULATION_STDEV)
             add_annotation(self.population_stdev, self)
 
     def _initialize(self):
-        self.shift_init.initialize(self.shift, self.rng)
-        if not self.mean_only:
+        if is_shared_variable(self.shift):
```

Comment:
Yeah, I should probably add a comment about that.

On Wed, Jun 1, 2016, 3:33 PM dmitriy-serdyuk <notifications@github.com>
wrote:

> In blocks/bricks/bn.py
> <https://github.com/mila-udem/blocks/pull/1104#discussion_r65427124>:
>
> >              self.population_stdev = shared_floatx(numpy.ones(var_dim),
> >                                                    name='population_stdev',
> >                                                    broadcastable=broadcastable)
> >              add_role(self.population_stdev, BATCH_NORM_POPULATION_STDEV)
> >              add_annotation(self.population_stdev, self)
> >
> >      def _initialize(self):
> > -        self.shift_init.initialize(self.shift, self.rng)
> > -        if not self.mean_only:
> > +        if is_shared_variable(self.shift):
>
> Ah, I see.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mila-udem/blocks/pull/1104/files/75bc90713ee7964b9aa8dfe92fbddf66936f63fe#r65427124>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADrLi-H5txTQ1nVVg8E2abgJxqWrO0mks5qHd54gaJpZM4Ir0Hx>
> .
>